src/Persistence_matrix/include/gudhi/Persistence_matrix/columns/unordered_set_column.h

```diff
@@ -0,0 +1,1008 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2022-24 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/**
+ * @file unordered_set_column.h
+ * @author Hannah Schreiber
+ * @brief Contains the @ref Unordered_set_column class.
+ * Also defines the std::hash method for @ref Unordered_set_column.
+ */
+
+#ifndef PM_UNORDERED_SET_COLUMN_H
+#define PM_UNORDERED_SET_COLUMN_H
+
+#include <vector>
+#include <stdexcept>
+#include <type_traits>
+#include <unordered_set>
```

Comment:
In other parts of Gudhi, we had nice improvements using recent Boost, either with unordered_set or the flat version (see https://github.com/GUDHI/gudhi-devel/blob/a895401811e0aee6a183ebf135bdcb504a220e23/src/Subsampling/include/gudhi/choose_n_farthest_points.h#L18-L22 for instance).

Reply:
Indeed, it is quite faster. So I changed to the same strategy than above!

Reply:
For `boost::unordered_set`, the difference with `std::unordered_set` was not so visible, but `boost::unordered_flat_set` was clearly faster than both. At least on random filtrations.